File: 2022/python/day01.py
def solve(data):
    calories = []
    cal_count = 0
    for line in data:
        if line == "":
            calories.append(cal_count)
            cal_count = 0
        else:
            cal_count += int(line)
    calories.sort()
    return max(calories), sum(calories[-3:])


def parse_input():
    return [line.strip() for line in open("inputs/day01.txt", 'r')]


def main():
    print(solve(parse_input()))


if __name__ == "__main__":
    main()

File: 2022/rust/src/days/day01.rs
use crate::lib::solution::Solution;
use crate::lib::solution::SolutionPair;

use std::fs::read_to_string;
use std::time::Instant;

pub fn solve() -> SolutionPair {
    let input: String = read_to_string("inputs/day01.txt").unwrap();

    // Part One
    let mut start = Instant::now();
    let mut calories: Vec<u32> = input.split("\n\n").map(
        |ls| ls.split("\n").map(|x| x.parse::<u32>().unwrap_or_default()).sum()
    ).collect();
    calories.sort();
    let l = calories.len();

    let part_one = Solution::U32(calories[l - 1]);
    // Part One End
    let duration_one = start.elapsed();

    // Part Two
    start = Instant::now();

    let part_two = Solution::U32(calories[l - 3 .. l].iter().sum());
    // Part Two End
    let duration_two = start.elapsed();
    
    SolutionPair { part_one, part_two, duration_one, duration_two }
}
